src/Workspaces/Core/Portable/Shared/Utilities/SymbolEquivalenceComparer.GetHashCodeVisitor.cs

```diff
@@ -172,6 +172,11 @@ private int CombineHashCodes(INamedTypeSymbol x, int currentHash)
 
             private int CombineNamedTypeHashCode(INamedTypeSymbol x, int currentHash)
             {
+                if (x.IsTupleType)
+                {
+                    return CombineHashCodes(x.TupleElementTypes, currentHash, _symbolAggregator);
```

Comment:
As far as the IDE is concerned, underlying types are irrelevant.  We only want to consider the element types of the tuple.